modules/@apostrophecms/piece-type/ui/apos/components/AposImageCropper.vue

```diff
@@ -74,27 +98,276 @@ export default {
+
+      // For some reason, positioning the focal point at the very top of the image
+      // results in having a negative `top` value.
+      // Let's apply a margin of error for every edge to prevent having `null`
+      // focal point values when placing it at the image borders.
+      const MARGIN_OF_ERROR = 1;
+
+      const limits = {
+        left: -focalPointSize.halfWidth + MARGIN_OF_ERROR,
+        top: -focalPointSize.halfHeight + MARGIN_OF_ERROR,
+        right: focalPoint.offsetParent.clientWidth - focalPointSize.halfWidth - MARGIN_OF_ERROR,
+        bottom: focalPoint.offsetParent.clientHeight - focalPointSize.halfHeight - MARGIN_OF_ERROR
+      };
+
+      if (left < limits.left || top < limits.top || left > limits.right || top > limits.bottom) {
+        return;
+      };
+
+      focalPoint.style.left = `${left}px`;
+      focalPoint.style.top = `${top}px`;
+
+      this.updateFocalPointCoordinatesDebounced();
+    },
+    /**
+     * Remove events when releasing click on the focal point element.
+     * Revert CSS values after manipulation.
+     */
+    onFocalPointMouseUp() {
+      const { focalPoint } = this.$refs;
+
+      focalPoint.style.cursor = 'grab';
+      focalPoint.style.transitionDuration = '0.1s';
+
+      document.removeEventListener('mousemove', this.onFocalPointMouseMove);
+      document.removeEventListener('mouseup', this.onFocalPointMouseUp);
+    },
+    /**
+     * Place focal point at the position where the image has been clicked
+     * and update its new coordinates.
+     * Position it in relation of the current target offset
+     * in order to set the `left` and `top` properties accordingly.
+     * Do not place focal point when cropper is changing (move or resize)
+     * to keep a friendly user experience.
+     */
+    onImageClick (event) {
+      if (this.isChangingCropper) {
+        return;
+      }
+
+      const { focalPoint } = this.$refs;
+
+      const focalPointSize = this.getFocalPointSize();
+
+      const left = event.clientX - event.currentTarget.offsetLeft - event.currentTarget.offsetParent.offsetLeft - focalPointSize.halfWidth;
+      const top = event.clientY - event.currentTarget.offsetTop - event.currentTarget.offsetParent.offsetTop - focalPointSize.halfHeight;
+
+      focalPoint.style.left = `${left}px`;
+      focalPoint.style.top = `${top}px`;
+
+      this.updateFocalPointCoordinatesDebounced();
+    },
+    /**
+     * Place the focal point back to its position (inside the stencil),
+     * when resizing the screen so that the percentages
+     * relative to the stencil remain the same.
+     */
+    onScreenResize () {
+      this.storeStencilCoordinates();
+      this.placeFocalPointInStencil();
+    },
+    /**
+     * Update stencil and focal point coordinates
+     * after cropper has changed and emit its new coordinates.
+     */
+    handleCropperChange (coordinates) {
+      this.isChangingCropper = false;
```

Comment:
Shouldn't it be at the end of this function? after the actual changes?

Reply:
We need to have the information that the cropper _is_ changing.

Placing it before or after `this.handleCropperChangeDebounced(coordinates);` is quite the same.
We could set `this.isChangingCropper = false;` after the call to the debounced function but that's more understandable to set it before, as to show that it is set as soon as the cropper is changing.

In any case, we don't want to set `this.isChangingCropper = false;` inside the debounced function.